pkg/cluster/https.go

```diff
@@ -33,12 +33,14 @@ func (c *Cluster) newListener(ctx context.Context) (net.Listener, http.Handler,
 	return dynamiclistener.NewListener(tcp, storage, cert, key, dynamiclistener.Config{
 		CN:           version.Program,
 		Organization: []string{version.Program},
-		TLSConfig: tls.Config{
+		TLSConfig: &tls.Config{
 			ClientAuth:   tls.RequestClientCert,
 			MinVersion:   c.config.TLSMinVersion,
 			CipherSuites: c.config.TLSCipherSuites,
 		},
-		SANs: append(c.config.SANs, "localhost", "kubernetes", "kubernetes.default", "kubernetes.default.svc."+c.config.ClusterDomain),
+		SANs:                  append(c.config.SANs, "localhost", "kubernetes", "kubernetes.default", "kubernetes.default.svc."+c.config.ClusterDomain),
+		ExpirationDaysCheck:   90,
+		CloseConnOnCertChange: true,
```

Comment:
Won't this cause some surprises on what I assume would be certificate rotation?

Reply:
Rotation is pretty infrequent, what is more likely to trigger this is adding a new node to the cluster, or adding a new SAN to the `--tls-san` list and restarting a node in the cluster (since any server can update the secret, and the other servers will pick up the change).

I can remove this if it seems like it is causing or would cause problems.

Reply:
As per Slack conversation, I am going to do some additional testing to see if there's any negative side effect to this. I don't expect anything worse than a connection retry but it's good to confirm.

Reply:
After some testing, I decided to skip this. While it doesn't seem to cause any problems, there is some additional log spew as things reconnect. Also, due to how it's implemented, only the node modifying the cert closes its connections. The other nodes still pick up the new cert from the datastore, but they don't close active connections, which kind of negates any positive effects that this might have.